Makefile

```diff
@@ -1,12 +1,23 @@
 # Compiler flags
 CFLAGS := -Wall -Wextra -O3 -m64 -D'uint64_t=unsigned long long'
 
+# Give Selfie information about the host operating system
+ifeq ($(OS),Windows_NT)
+    CFLAGS += -D'HOST_OS=3ull'
+else
+    ifeq ($(shell uname -s),Darwin)
+        CFLAGS += -D'HOST_OS=2ull'
+    else
+        CFLAGS += -D'HOST_OS=1ull'
+    endif
+endif
+
 # Bootstrap selfie.c into selfie executable
 selfie: selfie.c
 	$(CC) $(CFLAGS) $< -o $@
 
 # 32-bit compiler flags
-32-BIT-CFLAGS := -Wall -Wextra -Wno-builtin-declaration-mismatch -O3 -m32 -D'uint64_t=unsigned long'
+32-BIT-CFLAGS := $(subst -D'uint64_t=unsigned long long', -D'uint64_t=unsigned long', $(subst -m64, -m32, $(CFLAGS))) -Wno-builtin-declaration-mismatch
```

Comment:
How about defining HOSTOS and then use that in CFLAGS and 32-BIT-CFLAGS? This substitution business here is just an overkill.

Reply:
I liked that, because it emphasises, that 32 bit flags are derived from 64 bit flags. But anyway I have no hard opinion on that, so I'll revert that change.